### PR TITLE
feat(fal_client): align upload lifecycle API with fal-js

### PR DESCRIPTION
**Pull request by efiop**
## Summary
- Add `lifecycle` support to `fal_client` upload APIs (sync + async) and propagate it through all upload paths, including multipart and repository fallbacks.
- Normalize `lifecycle.expiresIn` values (`never`, `immediate`, `1h`, `1d`, `7d`, `30d`, `1y`, or integer seconds) into upload lifecycle headers compatible with the controller/CDN payload format.
- Extend unit tests for lifecycle header propagation, `expiresIn` normalization, and multipart behavior; mock CDN client creation in multipart tests to keep them fully local.

### CODE REVIEW

**Comment by efiop** [comment]
@claude review

**Comment by efiop** [comment]
Addressed Claude review feedback in `b7c5a8d0`:

- removed redundant `_resolve_upload_lifecycle` wrapper and use `_normalize_upload_lifecycle` directly
- changed lifecycle header guard to `is not None`
- avoided eager lifecycle normalization in `upload_file` non-multipart path
- skip serializing empty ACL `rules` arrays
- validate integer `expires_in` to require values > 0
- added tests for `expires_in=\"never\"`, non-positive integer rejection, and empty ACL rules

Also kept the earlier Python runtime compatibility fix for type aliases in `3971d5c2`.

**Comment by efiop** [comment]
@claude review

**Comment by efiop** [comment]
Addressed the latest Claude review item in `abeffadb`:

- `_normalize_upload_lifecycle` now skips `initial_acl` when `StorageACL()` is empty
- added regression test: `test_sync_upload_lifecycle_omits_empty_acl_object`

Validation run:
- `pytest projects/fal_client/tests/unit/test_client.py -k lifecycle`
- `pre-commit run --all-files`

**Comment by efiop** [comment]
@claude review

File: projects/fal_client/src/fal_client/client.py [hunk 1]
```diff
@@ -82,6 +88,41 @@
     thread_name_prefix="FAL_CLIENT_EXECUTOR"
 )
 
+_UPLOAD_LIFECYCLE_EXPIRATION_VALUES: dict[str, int | None] = {
+    "never": 3153600000,  # 100 years
+    "immediate": None,
```
Comment (hanchchch) [inline projects/fal_client/src/fal_client/client.py]:
```suggestion
    "never": None,
    "immediate": 0,
```

File: projects/fal_client/src/fal_client/client.py [hunk 2]
```diff
@@ -82,6 +88,41 @@
...
+class StorageACL:
+    default: StorageACLDecision | None = None
+    rules: list[StorageACLRule] | None = None
+
+
+@dataclass(frozen=True)
+class StorageSettings:
+    """Lifecycle configuration for uploaded files.
+
+    Mirrors the fal-js `StorageSettings` model while exposing additional
```
Comment (hanchchch) [inline projects/fal_client/src/fal_client/client.py]: i think we don't have to mention about fal-js here

File: projects/fal_client/src/fal_client/client.py (hunk 1 above)
Comment (efiop) [reply projects/fal_client/src/fal_client/client.py]: thats how it got introduced in https://github.com/fal-ai/fal-js/pull/170/changes , trying to be in sync. Let me check if thse actually work though

Comment (efiop) [reply projects/fal_client/src/fal_client/client.py]: adjusted

**Comment by efiop** [comment]
Addressed the latest Claude review in `72a4191c`:

- added sync/async multipart `upload_file` lifecycle propagation tests:
  - `test_sync_upload_file_passes_lifecycle_to_multipart`
  - `test_async_upload_file_passes_lifecycle_to_multipart`
- documented `allow_io_storage` auto-inference behavior inline in `_normalize_upload_lifecycle` when `expires_in` is provided and no explicit override is set

Validation:
- `pytest projects/fal_client/tests/unit/test_client.py -k "upload_file_passes_lifecycle_to_multipart or upload_passes_lifecycle_to_multipart or lifecycle"`
- `pre-commit run --all-files`

**Comment by efiop** [comment]
@claude review

**Comment by efiop** [comment]
Addressed the latest Claude review in `0616d51e`:

- reject boolean lifecycle values (`expires_in=True/False`) in `_get_upload_expiration_duration_seconds`
- align error wording to Python snake_case (`expires_in`) for both integer and unsupported-string validation errors
- export `ObjectExpiration` and `StorageACLDecision` from `fal_client.__init__`
- add regression tests for negative integer, boolean, and unsupported string lifecycle values
- add inline clarification for why `immediate` maps to a small positive TTL

Validation:
- `pytest projects/fal_client/tests/unit/test_client.py -q`
- `pre-commit run --all-files`

**Comment by efiop** [comment]
@claude review

**Comment by efiop** [comment]
@claude review

**Comment by efiop** [comment]
@claude review

**Review by yemreck — APPROVED** [review]
